functorch/experimental/_cond.py

```diff
@@ -56,12 +56,35 @@ def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
 
     flat_true_outs, _ = pytree.tree_flatten(true_outs)
     flat_false_outs, _ = pytree.tree_flatten(false_outs)
-    assert(len(flat_true_outs) == len(flat_false_outs))
+    if len(flat_true_outs) != len(flat_false_outs):
+        raise UserError(
+            UserErrorType.COND_OP_RESTRICTION,
+            "Expect true branch and false branch return with same length but got\n"
```

Comment:
This message is not very clear.
1. Grammatic fix: "Expected ... but got ..."
2. "return with same length" -> "to return same number of outputs"
3. No point emitting `true_outs` and `false_outs` (could be large!), just emit their `len()`?